packages/components/src/components/SearchBar.vue

```diff
@@ -185,78 +182,125 @@ export default {
     },
-    onWindowClick(event) {
+    incrementSelectedSuggestion() {
+      if (this.selectedSuggestion !== false && this.selectedSuggestion !== 0) {
+        this.selectedSuggestion -= 1;
+      } else {
+        this.selectedSuggestion = this.suggestionsList.length - 1;
+      }
+    },
+    decrementSelectedSuggestion() {
       if (
-        !getEventTarget(event.target, this.$refs.form) &&
-        this.$refs.input !== window.document.activeElement
+        this.selectedSuggestion !== false &&
+        this.selectedSuggestion !== this.suggestionsList.length - 1
       ) {
-        this.showSuggestions = false;
+        this.selectedSuggestion += 1;
+      } else {
+        this.selectedSuggestion = 0;
+      }
+    },
+    onWindowClick(event) {
+      if (!getEventTarget(event.target, this.$refs.form)) {
+        if (this.$refs.input !== window.document.activeElement) this.showSuggestions = false;
+
+        if (
+          !event.target.classList.contains('search-bar__suggestions-item') &&
+          !this.searchValue.endsWith(' ')
+        )
+          this.searchValue += ' ';
       }
     },
     onWindowKeyUp(event) {
       if (this.$refs.input !== window.document.activeElement) {
         return;
       }
+      if (this.keyDownTimeoutId !== undefined) {
+        clearTimeout(this.keyDownTimeoutId);
+        this.keyDownTimeoutId = undefined;
+      }
+      if (this.keyDownIntervalId !== undefined) {
+        clearInterval(this.keyDownIntervalId);
+        this.keyDownIntervalId = undefined;
+        return;
+      }
 
       switch (event.key) {
         case 'Escape':
           this.showSuggestions = false;
           this.$refs.input.blur();
+          if (!this.searchValue.endsWith(' ')) this.searchValue += ' ';
           break;
         case 'ArrowDown':
-          if (
-            this.selectedSuggestion !== false &&
-            this.selectedSuggestion !== this.suggestionsList.length - 1
-          ) {
-            this.selectedSuggestion += 1;
-          } else {
-            this.selectedSuggestion = 0;
-          }
+          this.decrementSelectedSuggestion();
           break;
         case 'ArrowUp':
-          if (this.selectedSuggestion !== false && this.selectedSuggestion !== 0) {
-            this.selectedSuggestion -= 1;
-          } else {
-            this.selectedSuggestion = this.suggestionsList.length - 1;
-          }
+          this.incrementSelectedSuggestion();
           break;
         default:
           this.selectedSuggestion = false;
           break;
       }
     },
+    onWindowKeyDown(event) {
+      if (
+        this.$refs.input !== window.document.activeElement ||
+        this.keyDownTimeoutId !== undefined
+      ) {
+        return;
+      }
+
+      this.keyDownTimeoutId = setTimeout(() => {
+        switch (event.key) {
+          case 'ArrowDown':
+            this.keyDownIntervalId = setInterval(this.decrementSelectedSuggestion.bind(this), 150);
```

Comment:
I'm not sure what you mean, can you elaborate?

Reply:
Normally when you hold down a key the repetition rate is controlled by the system. This allows the user to adjust it which can be very important for people with movement disabilities (eg. when they need to take longer to react). Here you're hardcoding it to 150 ms, which is I guess on the annoyingly slow side to me, but on the other hand can be too fast for some people. It'd be great if we could instead use the system key repetition instead of rolling our own.

Reply:
I did some experiments and I noticed you don't need to roll your own repetition. Just handle `keydown` and the browser will automatically fire repeated events. This will respect the user expectations and settings.

Reply:
Okay, thanks for your help on this, I changed it to work with the system repetition :+1: